### ISSUE TITLE
Jboss EAP 7

### ISSUE THREAD

**Issue by megalexis** [Impetus/kundera#883]
Unable to use EJB resources with resteasy using container managed transactions.
Is there any documentation or example project on how to properly configure this?

my persistence.xml:
`<?xml version="1.0" encoding="UTF-8"?>
<persistence version="2.1" xmlns="http://xmlns.jcp.org/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
    <persistence-unit name="velocityProfileServiceJPA">
    <provider>com.impetus.kundera.KunderaPersistence</provider>
    <class>com.test.services.velocity.profile.domain.Column</class>
    <class>com.test.services.velocity.profile.domain.Data</class>
    <class>com.test.services.velocity.profile.domain.Profile</class>
    <class>com.test.services.velocity.profile.domain.View</class>
    <exclude-unlisted-classes>true</exclude-unlisted-classes>           
        <properties>
            <property name="jboss.as.jpa.managed" value="false" />
            <property name="kundera.nodes" value="localhost"/>
            <property name="kundera.port" value="9160"/>
            <property name="kundera.keyspace" value="vps"/>
            <property name="kundera.dialect" value="cassandra"/>
            <property name="kundera.ddl.auto.prepare" value="create" />
            <property name="kundera.client.lookup.class" value="com.impetus.client.cassandra.thrift.ThriftClientFactory" />         
        </properties>
    </persistence-unit>
</persistence>`

error:
`01:34:40,932 ERROR [org.jboss.as.controller.management-operation] (DeploymentScanner-threads - 2) WFLYCTL0013: Operation ("full-replace-deployment") failed - address: ([]) - failure description: {"WFLYCTL0180: Services with missing/unavailable dependencies" => [
    "jboss.naming.context.java.comp.\"velocityProfileServiceEAR-0.0.1-SNAPSHOT\".\"velocityProfileServiceEJB-0.0.1-SNAPSHOT\".ProfileResourceBean.env.velocityProfileServiceJPA is missing [jboss.persistenceunit.\"velocityProfileServiceEAR-0.0.1-SNAPSHOT.ear#velocityProfileServiceJPA\"]",
    "jboss.naming.context.java.comp.\"velocityProfileServiceEAR-0.0.1-SNAPSHOT\".\"velocityProfileServiceEJB-0.0.1-SNAPSHOT\".DataResourceBean.env.velocityProfileServiceJPA is missing [jboss.persistenceunit.\"velocityProfileServiceEAR-0.0.1-SNAPSHOT.ear#velocityProfileServiceJPA\"]"
]}`

Seems to almost work by commenting out:

`<property name="jboss.as.jpa.managed" value="false" />`

and annotating the beans with:
`@TransactionManagement(TransactionManagementType.BEAN)`

but fails with exception:

`jboss error Please bind [com.impetus.kundera.persistence.jta.KunderaJTAUserTransaction] for :{java:comp/UserTransaction} lookupclass org.jboss.tm.usertx.client.ServerVMClientUserTransaction
`
I have not figured out a way to successfully bind this on Jboss EAP 7

**Comment by devender-yadav** [comment]
@megalexis 

Have you seen [Kundera with JBoss](https://github.com/impetus-opensource/Kundera/wiki/Using-Kundera-with-Jboss) wiki. 

-Dev

**Comment by megalexis** [comment]
Yes looked at this but we want to use pure EJBs and container-managed transactions with resteasy. We have to use the JEE stack. A spring app context is not an option for us. Looks like some progress was made [https://github.com/impetus-opensource/Kundera/issues/344](https://github.com/impetus-opensource/Kundera/issues/344) but it was not documented.
